server/routes/api.js

```diff
@@ -59,9 +59,9 @@ router.get("/metrics", metricsController.getMetrics);
 
 router.get("/metrics/:id", metricsController.getMetric);
 
-router.post("/metrics", metricsController.createMetric);
+router.post("/metrics", metricsController.validateQuery, metricsController.createMetric);
 
-router.put("/metrics/:id", metricsController.editMetric);
+router.put("/metrics/:id", metricsController.validateQuery, metricsController.editMetric);
 
 router.delete("/metrics/:id", metricsController.deleteMetric);
```

Comment:
Ooh interesting, I wonder if it's because metric_id is a primary key in the `experiment_metrics` table. I don't know how to fix that so yeah I can just make an `is_deleted` column instead. 

I think it's fine to delete metrics that are being used in current experiments, I assume they know what they're doing and want it deleted for a reason. We could add an additional warning message though, like "This metric is currently being used in an experiment, are you sure you want to continue?"